### ISSUE TITLE
Watching network drive '\\server\folder' does not work

### ISSUE THREAD

**Issue by Zireael** [paulmillr/chokidar#895]
Hi
I have a problem where chokidar does not report changes to network shared folder, i.e. `\\\\file.server.net\\dev\\7500`. Using `fs.writeFile()` on that folder works fine and a file gets created by Node. I'm under Win10.
My config is below. I tried both with `usePolling` true and false, no change.
I don't get any errors, just changes to network folder don't get reported. Watching a folder on local drive C:\db works fine. Watching that network drive using drive letter path format actually works too `s:\\dev\\7500`. I'd prefer to use server path notation in case drive mapping letters get changed in future, or some users in the company have different letters mapped.

Do you see anything amiss in my setup?

```
fs.writeFile('\\\\file.server.net\\dev\\7500\\test.txt', 'this file gets created OK', (err) => {
  if (err) throw err
})

const chokidarOptions = {
  persistent: true,
  // ignored: "*.txt",
  ignoreInitial: false,
  followSymlinks: true,
  cwd: ".", 
  disableGlobbing: false,
  usePolling: true, // tried with false as well
  interval: 1000,
  binaryInterval: 3000,
  alwaysStat: false,
  depth: 99,
  awaitWriteFinish: {
    stabilityThreshold: 2000,
    pollInterval: 100,
  },
  ignorePermissionErrors: true, // watch files that don't have read permissions
  atomic: true, // or a custom 'atomicity delay', in milliseconds (default 100)
};

const optionsFileMonitor = {
  watchFolders: ["\\\\file.server.net\\dev\\7500", "s:\\dev\\7500", "C:/db"]
};

const fileMonitor = chokidar
  .watch(optionsFileMonitor.watchFolders, chokidarOptions)
  .on("all", (event, epath) => {
    console.log(event, path.resolve(epath));
})
```

**Comment by puglisij** [comment]
I have the same issue watching a directory on a CIFS drive.   
For example //cifs/foo/bar 
I get no reported file changes.  I DO get changes reported when using the native  fs.watch()  
It looks like there may be an issue with path normalization in chokidar?

**Comment by SutharMukesh** [comment]
I too had this problem  
downgrading to chowkidar@3.1.1 solves for me.

**Comment by valdiss** [comment]
I have the exact same issue as Zireal, watching a folder on our network drive does not work on Windows 10.
I tried with fs.watch, it does work.

I downgraded to 3.1.1, it does not fix the issue.

**Comment by rpsirois** [comment]
Bump

**Comment by kishore5214-1** [comment]
Changing to @3.1.1 worked for me

**Comment by jaredbaszler** [comment]
Changing to @3.1.1 worked for me as well.  I was on 3.4.0 when it wasn't working.

**Comment by nicograef** [comment]
> Changing to @3.1.1 worked for me as well. I was on 3.4.0 when it wasn't working.

+1 here 👋

**Comment by paulmillr** [comment]
No status. Feel free to debug this and provide a pull req.

**Comment by whyboris** [comment]
Confirming version `3.4.0` does not work on my network files:
`\\StoragePC\DrivePool\Movies`
When pointing to this directory, _chokidar_ does not find any files (even though they are there and are accessible using `fs`) 😢

**Comment by whyboris** [comment]
I did some debugging, I think I see the issue:
https://github.com/paulmillr/chokidar/blob/master/index.js#L395
When the `.add()` happens, the first thing executed is `let paths = unifyPaths(paths_);`
unifyPaths eats the first `/` in the address, so:
`\\StoragePC\DrivePool\Movies` goes in (even `//StoragePC/DrivePool/Movies` works here)
and `unifyPaths` returns
`/StoragePC/DrivePool/Movies` ⚠️ (note the first `/` is removed) 👀

**Comment by whyboris** [comment]
@paulmillr -- thank you for your excellent library 🙇 

I feel like I've correctly identified where the problem happens (replacing double slashes with a single slash) ✅ 

I suspect one solution is to somehow ignore the double slashes if they occur as the first characters 🤔 

I don't know what would be the best way to fix this without breaking something for everyone else. 

I'm happy to make a PR if you suggest how I should handle it 😁  🤝  😄

**Comment by jaredbaszler** [comment]
> Changing to @3.1.1 worked for me as well. I was on 3.4.0 when it wasn't working.

This worked for me to get it to listen to a network folder correctly. But then I eventually had to shut it off and reroute things to a local project folder. The reason being once I had this monitoring a network folder my database called with the OracleDB package slowed WAY down.  Simple Select queries were taking around 10 seconds and once I changed Chokidar to monitor a network folder I got the queries back down to less than .1 seconds. I'm not sure why but just passing along my experience.

**Comment by whyboris** [comment]
👋 @paulmillr -- please let me know if the solution I propose to this problem (see above) is acceptable, namely:

**do not replace double slash `//` with single slash `/` if it occurs at the beginning of the string**

I would like to create a PR, but wanted confirmation that this is an acceptable (mergeable) solution or if you prefer something else 🙇

**Comment by paulmillr** [comment]
@whyboris try doing this and see if tests fail on any platform

**Comment by whyboris** [comment]
PR to solve the problem 🤞 #1025 😁

**Comment by jmeinke** [comment]
@paulmillr I think this issue should not be closed while there is no new chokidar release available via npm that includes the fix provided by @whyboris.

**Comment by oumad** [comment]
I'm still having this issue, I hope the PR that fixes it will be available in the next chokidar update

**Comment by oumad** [comment]
> @oumad -- seems like the fix is available since release `3.4.2`
> https://github.com/paulmillr/chokidar/releases/tag/3.4.2
> Have you tested?

Interesting, then I don't know why for some reason I get no event fired when I try watching a directory in this format `\\server\path\to\folder\*.{extentions}`
But no problem when I watch the same directory like this :
`S:/path/to/folder/*.{extentions}`
regardless of usePolling set to true or false.

**Comment by TierK** [comment]
have the same issue with chokidar 3.5.3

In local folders, it works perfectly. But it doesn't work with remote folders with the same functions
